TrainingGym/env/SumoEnvParallel.py

```diff
@@ -17,21 +18,30 @@
 else:
     sys.exit("please declare environment variable 'SUMO_HOME'")
 from sumolib import checkBinary
+from traci._trafficlight import Phase, Logic
 
 with open('global_config.json') as global_json_file:
-    local_config_path = json.load(global_json_file)['config_path']
+    global_config_params = json.load(global_json_file)
+    local_config_path = global_config_params['config_path']
 with open(f'Scenarios/{local_config_path}') as json_file:
     config_params = json.load(json_file)
 
+# Constants
+YELLOW_LENGTH = 4.2  # seconds
+RED_LENGTH = 1.8  # seconds
+STEP_LENGTH = 1.0  # seconds
+
 # Load config
 controlled_lights = config_params['controlled_lights']
-uncontrolled_lights = config_params['uncontrolled_lights']
+for i in range(len(controlled_lights) - 1, -1, -1):
```

Comment:
I'm not sure, since we're deleting items from the list the index will change, so we have to traverse it backwards. Or at least that's what I thought. maybe python is different/

Reply:
Ah yeah, you're right.